controller/QtiCreator.php

```diff
@@ -369,4 +371,31 @@ private function validateXmlInput(string $xml)
         }
         \tao_helpers_Xml::getSimpleXml($xml);
     }
+
+    private function getItemIdentifier(core_kernel_classes_Resource $item): ?string
+    {
+        $identifier = $this->getIdentifierGenerator()->generate([
+            IdentifierGenerator::OPTION_RESOURCE_ID => $item->getUri(),
+        ]);
+
+        if (!$this->getFeatureFlagChecker()->isEnabled('FEATURE_FLAG_UNIQUE_NUMERIC_QTI_IDENTIFIER')) {
+            return $identifier;
+        }
+
+        $uniqueId = $item->getOnePropertyValue($this->getProperty(TaoOntology::PROPERTY_UNIQUE_IDENTIFIER));
+
+        return $uniqueId
+            ? $uniqueId->literal
+            : $identifier;
```

Comment:
- [ ] @shpran Should not we ask if the Item has the `TaoOntology::PROPERTY_UNIQUE_IDENTIFIER` already first? In the clone, we should set the proper QTI and Metadata for unique identifier, why would we generate it here?
- [ ] Does it make sense to move this logic to `Service::singleton()->getDataItemByRdfItem` and remove all the logic from this controller? I think it is more clean and controller will not deal with business logic

```suggestion
        if ($this->getFeatureFlagChecker()->isEnabled('FEATURE_FLAG_UNIQUE_NUMERIC_QTI_IDENTIFIER')) {
            $uniqueId = $item->getOnePropertyValue($this->getProperty(TaoOntology::PROPERTY_UNIQUE_IDENTIFIER));
        
            if ($uniqueId) {
                return $uniqueId->literal;
            }
        }

        return $this->getIdentifierGenerator()->generate([
            IdentifierGenerator::OPTION_RESOURCE_ID => $item->getUri(),
        ]);
```

Reply:
This part of code was done only to send ID to fronted before XML creation and will be removed, because we do not need synchronization between property and QTI ID anymore.

Reply:
We still need the synchronization, for the moment